packages/x-charts/src/context/SurfacePropsProvider/SurfacePropsProvider.tsx

```diff
@@ -0,0 +1,26 @@
+'use client';
+import * as React from 'react';
+import { SurfacePropsContext } from './SurfacePropsContext';
+import { SurfacePropsProviderProps } from './SurfaceProps.types';
+
+const SurfacePropsProvider = React.forwardRef(function SurfacePropsProvider(
+  props: SurfacePropsProviderProps,
+  ref: React.Ref<SVGSVGElement>,
+) {
+  const { children, ...other } = props;
+
+  const value = React.useMemo(
+    () => ({
+      isInitialized: true,
+      data: {
+        ...other,
+        ref,
+      },
+    }),
+    [other, ref],
+  );
```

Comment:
I thaught you would complitely remove this provider. If we look at `ChartsSurfaceProps` attribute, most of them can be passed directly by the user or propagated by the `ChartContainer` (the `useChartContainerProps` could get an additional output `chartsSurfaceProps`):

- className
- title
- desc
- sx
- children

On the same way that it's done on the gauge chart

https://github.com/mui/mui-x/blob/d1e72d3afafb6ae172d32ac94913969cc0ff683f/packages/x-charts/src/Gauge/GaugeContainer.tsx#L107


The only attributes that for me make sense to pass to the DataProvider is `disableAxisListener` because it's not strictly related to the SVG so it could be handled outside of it